### ISSUE TITLE
More complex `no-invalid-interactive` checking

### ISSUE THREAD

**Issue by Yelinz** [ember-template-lint/ember-template-lint#3072]
I stumbled into the [`no-invalid-interactive`](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-invalid-interactive.md) rule for selectable table rows. The table component can display a checkbox for selection. If it does the whole row should be clickable for better UX.
This leads to the rule being triggered because role is not directly set. As it relies on an if depending on the argument. 

```hbs
<tr
  {{on "click" (fn (optional @selectRow) entry)}}
  role={{if @selectRow "button" "row"}}
>
```

I was wondering if this would be worth to add more complex rules or just accept `{{! template-lint-disable no-invalid-interactive }}` before the element.

**Comment by sukima** [comment]
I too was wondering how to have event bubbling/delegation when this rule is in place. Having to disable every time we wrap dynamic content is kinda a show stopper for every developer using an in house design system. I think the only way to avoid the poor developer experience with this is to make a custom modifier to side step the linter.

I wish this rule had a `{{on "…" … delegated=true}}` bypass or something.

**Comment by sukima** [comment]
Example use case:

```hbs
<div {{on "click" this.handleButtonClicks}}>
  {{format-strings-with-buttons-in-them
    (t "INTL-string-with-custom-button-tags")
  }}
</div>
```

Might generate some DOM like:

```html
<div>
  This is a sample string with <button type="button">an action</button> in it.
</div>
```

Since the sentence is dynamically generated and place into the DOM via a SafeString (yes sanitation is performed) we would want to capture the click event one level higher but due to linting we cannot or we have to document in educate anyone who uses the `format-strings-with-buttons-in-them` helper to also add `{{! template-lint-disable no-invalid-interactive}}` which is weird, annoying, and very error prone.

I would like to petition that event delegation/bubbling is an acceptable use of placing interactivity on a non-interactive element!

**Comment by MelSumner** [comment]
Per ARIA 1.2 (and maintained in the draft for 1.3):

> If the tabular container maintains a selection state, provides its own two-dimensional navigation, or allows the user to rearrange or otherwise manipulate its contents or the display thereof, authors SHOULD use [grid](https://w3c.github.io/aria/#grid) or [treegrid](https://w3c.github.io/aria/#treegrid) instead.

**Comment by MelSumner** [comment]
Also: 
> If it does the whole row should be clickable for better UX.

This may be a wide-spread practice but it is still not WCAG conformant. 

I think if you decide to do this and do not have WCAG conformance as a requirement then you should inline disable the rule. The rule itself should not allow for this.

**Comment by johanrd** [comment]
yeah, I basically did this:

```hbs
<tr
  {{! template-lint-disable no-invalid-interactive}}
  {{! just a convenience fallback interaction for the already accessible <input type="checkbox"> in the first column}}
  {{on 'click' (fn this.handleRowClick row)}}
/>
```

**Comment by sukima** [comment]
Just out of curiosity how does JavaScript adding an event handler on an element or not affect accessibility? I had thought screen readers and other web scrappers inspected the elements not the JS managing them.

For example I would not expect these two to be any different from an accessibility perspective. Both perform the action and both have an accessible button for the interaction:

```html
<div>
  <button></button> <-- handler attached here
</div>
```

```html
<div> <-- handler attached here
  <button></button>
</div>
```

I guess I'm asking are **all** websites that do this incapable of being compliant?

```js
document.body.addEventListener((event) => {
  // quick n' dirty event delegation
  sendAction(event.target.dataset.action);
});
```

Perhaps in some cases someone _might_ forget to include an accessible element within. But if a dev knows it is accessible even if the handler is higher up the tree is that still OK? If so, I presume this is a case for a lint-disable directive.
